core/src/main/java/com/google/zxing/client/result/ResultParser.java

```diff
@@ -228,7 +228,10 @@ static String urlDecode(String encoded) {
             element = element.trim();
           }
           if (!element.isEmpty()) {
-            matches.add(element);
+            List<String> mailAdresses = element.split(','); // split the email adresses by comma 
+            for(String mail : mailAdresses){
```

Comment:
use here forEach is more clear, in my opinion
mailAdresses.forEach(mailAddres->matches.add(mailAddres));